src/lifx/light.js

```diff
@@ -529,4 +529,162 @@ Light.prototype.colorZones = function(startIndex, endIndex, hue, saturation, bri
   this.client.send(packetObj, callback);
 };
 
+/**
+ * Requests tile getDeviceChain 701
+ * @param {Function} callback a function to accept the data
+ */
+Light.prototype.getDeviceChain = function(callback) {
+  validate.callback(callback, 'light getDeviceChain method');
+
+  const packetObj = packet.create('getDeviceChain', {}, this.client.source);
+  packetObj.target = this.id;
+  const sqnNumber = this.client.send(packetObj);
+  this.client.addMessageHandler('stateDeviceChain', function(err, msg) {
+    if (err) {
+      return callback(err, null);
+    }
+    return callback(null, msg);
+  }, sqnNumber);
+};
+
+/**
+ * Sets Tile Position
+ * @example light.setUserPosition(0, 12, 13, 0, () => {})
+ * @param {Number} tileIndex	unsigned 8-bit integer
+ * @param {Number} userX	32-bit float
+ * @param {Number} userY	32-bit float
+ * @param {Number} reserved	16-bit integer
+ * @param {Function} callback called when light did receive message
+ */
+Light.prototype.setUserPosition = function(tileIndex, userX, userY, reserved, callback) {
+  if (tileIndex === undefined || typeof tileIndex !== 'number') {
+    throw new TypeError('LIFX light setUserPosition tileIndex to be a number');
+  }
+  if (userX === undefined || typeof userX !== 'number') {
+    throw new TypeError('LIFX light setUserPosition userX to be a number');
+  }
+  if (userY === undefined || typeof userY !== 'number') {
+    throw new TypeError('LIFX light setUserPosition userY to be a number');
+  }
+  validate.optionalCallback(callback, 'light setUserPosition method');
+
+  const packetObj = packet.create('setUserPositiion', {
+    tileIndex,
+    userX,
+    userY,
+    reserved: reserved || 0
+  }, this.client.source);
+  packetObj.target = this.id;
+  this.client.send(packetObj, callback);
+};
+
+/**
+ * Requests tile GetTileState64 707
+ *
+ * Get the state of 64 pixels in the tile in a rectangle that has
+ * a starting point and width.
+ * The tileIndex is used to control the starting tile in the chain
+ * and length is used to get the state of that many tiles beginning
+ * from the tileIndex. This will result in a separate response from
+ * each tile.
+ * For the LIFX Tile it really only makes sense to set x and y to
+ * zero, and width to 8.
```

Comment:
If this comment holds, would it then make sense to move `x`, `y` and `width` as the last parameters and give them a default value using the `function(foo, bar = 'bar')` syntax? If the `babel` preset we have now does not support that, we ought to fix that first. If there's a specific reason for the current order of the parameters, please let me know.

I wouldn't want to rush this change as people might start creating pretty complex animations etc., after which it would be quite rude to change it.

Reply:
i change the api of set/get TileState64 to an option prop style
where the defaults are applied safely.